### PR TITLE
PYIC-3187: Update `BuildUserIdentityDetails` to fetch and return CIMIT VC alongside the CRI VCs

### PR DESCRIPTION
**Pull request by TalatAydinGDS**
### What changed

Update `BuildUserIdentityDetails` to fetch and return CIMIT VC alongside the CRI VCs

### Why did it change

added the CiMit's VC to the UserIdentity's VC list

### Issue tracking

- [PYIC-3187](https://govukverify.atlassian.net/browse/PYIC-3187)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[PYIC-3187]: https://govukverify.atlassian.net/browse/PYIC-3187?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by lwebb-iig — DISMISSED** [review]
LGTM

File: lambdas/build-user-identity/src/main/java/uk/gov/di/ipv/core/builduseridentity/BuildUserIdentityHandler.java [hunk 1]
```diff
@@ -138,6 +146,15 @@ public APIGatewayProxyResponseEvent handleRequest(
                             ipvSessionItem.getVot(),
                             ipvSessionItem.getCurrentVcStatuses());
 
+            if (configService.enabled(CoreFeatureFlag.USE_CONTRA_INDICATOR_VC)) {
+                String signedCiMitJwt =
+                        ciMitService.getContraIndicatorsVcAsJwtString(
+                                userId,
+                                clientOAuthSessionItem.getGovukSigninJourneyId(),
+                                ipAddress);
```
Comment (therealslimsladey) [inline lambdas/build-user-identity/src/main/java/uk/gov/di/ipv/core/builduseridentity/BuildUserIdentityHandler.java]: Looks like `ipAddress` is not available for `BuildUserIdentity` - see commit `f1426ef6`. To be consistent with the audit call we should pass null

**Review by rpayal — DISMISSED** [review]
LGTM

File: lib/src/main/java/uk/gov/di/ipv/core/library/service/CiMitService.java [hunk 2]
```diff
@@ -149,6 +149,23 @@ public List<ContraIndicatorItem> getCIs(
...
             String userId, String govukSigninJourneyId, String ipAddress)
             throws CiRetrievalException {
+        SignedJWT ciSignedJWT = getContraIndicatorsJWT(userId, govukSigninJourneyId, ipAddress);
+        ContraIndicatorEvidenceDto contraIndicatorEvidence =
+                parseContraIndicatorEvidence(ciSignedJWT);
+
+        return mapToContraIndications(contraIndicatorEvidence);
+    }
+
+    public String getContraIndicatorsVcAsJwtString(
```
Comment (therealslimsladey) [inline lib/src/main/java/uk/gov/di/ipv/core/library/service/CiMitService.java]: Or just make `getContraIndicatorsJWT` (which I'd rename to `getContraIndicatorsVCJwt`) public and let BuildUserIdentity serialise it to string?

**Review by therealslimsladey — CHANGES_REQUESTED** [review]
looks good, just a couple of minor comments

File: lambdas/build-user-identity/src/main/java/uk/gov/di/ipv/core/builduseridentity/BuildUserIdentityHandler.java (hunk 1 above)
Comment (TalatAydinGDS) [reply lambdas/build-user-identity/src/main/java/uk/gov/di/ipv/core/builduseridentity/BuildUserIdentityHandler.java]: Thanks catch this point. Done.

File: lib/src/main/java/uk/gov/di/ipv/core/library/service/CiMitService.java (hunk 2 above)
Comment (TalatAydinGDS) [reply lib/src/main/java/uk/gov/di/ipv/core/library/service/CiMitService.java]: Done.

**Review by therealslimsladey — APPROVED** [review]
lgtm

**Pull request merged**